api/lib/rest-to-tx-converter.js

```diff
@@ -19,28 +19,15 @@ function RestToTxConverter() {}
  *  @param {ripple-lib Transaction} transaction
  */
 RestToTxConverter.prototype.convert = function(payment, callback) {
-  function isSendMaxRequired() {
-    var src = payment.source_account;
-
+  function isSendMaxAllowed() {
     var srcAmt = payment.source_amount;
     var dstAmt = payment.destination_amount;
 
     // Don't set SendMax for XRP->XRP payment
-    if (!srcAmt || srcAmt.currency === 'XRP' && dstAmt.currency === 'XRP') {
-      return false;
-    }
-
-      // only set send max when:
-      // - source and destination currencies are same
-      //   and source issuer is not source account
-      // - source amount and destination issuers are different
-      //
-    if (srcAmt.currency === dstAmt.currency) {
-      if (srcAmt.issuer !== src && srcAmt.issuer !== dstAmt.issuer) {
-        return true;
-      }
-    }
-    return false;
+    // temREDUNDANT_SEND_MAX removed in:
+    // https://github.com/ripple/rippled/commit/
+    //  c522ffa6db2648f1d8a987843e7feabf1a0b7de8/
+    return srcAmt && srcAmt.currency !== 'XRP' && dstAmt.currency !== 'XRP';
```

Comment:
SendMax only allowed for anything that's not XRP to XRP (and includes a source amount)